### PR TITLE
Module modernisation

### LINKED ISSUE #24: LICENSE file missing from master
The LICENSE file is missing from your master branch. It's referenced from your README.md, but the file itself is missing.

p.s. Thx for making this available under a permissive license :)

### PR DESCRIPTION
**Pull request by joe-niland**
## what
* add context
* add licence
* upgrade for TF 0.14
* add tests
* add most recent github workflows

## why
* allow proper management of PRs
* allow use with TF 0.13+

## references
* closes #24

### CODE REVIEW

File: examples/complete/provider.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+provider "aws" {
+  region = var.region
+
+  # Make it faster by skipping something
```
Comment (aknysh) [inline examples/complete/provider.tf]: I'd delete these, we don't use these anymore, it's fast w/o them :)

File: examples/complete/main.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  skip_final_snapshot  = "true"
+}
+
+module "rds_alarms" {
+  source         = "../../"
+  db_instance_id = aws_db_instance.default.id
+  context        = module.this.context
+}
+
+
```
Comment (aknysh) [inline examples/complete/main.tf]: delete the second empty line

File: examples/complete/variables.tf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+variable "region" {
+  type        = string
+  description = "AWS region"
+  default     = "us-east-2"
+}
+
+variable "marbot_endpoint_id" {
```
Comment (aknysh) [inline examples/complete/variables.tf]: why this variable not in `examples/marbot` example?
and what is `marbot` ? :)

**Review by aknysh — COMMENTED** [review]
thanks @joe-niland, LGTM, just a few comments

File: examples/complete/variables.tf (hunk 3 above)
Comment (joe-niland) [reply examples/complete/variables.tf]: I think this was due to my error when reorganising files.

The module had this marbot example already, but my opinion is it's not that useful.

@aknysh do you agree we can remove this example?

Marbot is a chat bot service for AWS monitoring from cloudonaut.
https://marbot.io/

Comment (aknysh) [reply examples/complete/variables.tf]: @joe-niland let's remove that example (it's confusing)

**Comment by joe-niland** [comment]
@aknysh hopefully all addressed

**Review by aknysh — APPROVED** [review]

**Comment by aknysh** [comment]
@joe-niland  approved, thanks

**Pull request merged**